cpp/src/Ice/Exception.cpp

```diff
@@ -539,6 +461,54 @@ Ice::Exception::ice_stackTrace() const
     return getStackTrace(*_stackFrames);
 }
 
+void
+Ice::Exception::ice_enableStackTraceCollection()
+{
+    lock_guard lock(globalMutex);
+#if defined(ICE_DBGHELP)
+    if (process)
+    {
+        // Already initialized, just refresh.
+        if (!SymRefreshModuleList(process))
+        {
+            // TODO: SymRefreshModuleList occasionally fails with error code 3221225476; we retry once in this case.
+            // Note that calling GetLastError() does not reset the last error.
+            if (GetLastError() != 3221225476 || !SymRefreshModuleList(process))
+            {
+                throw std::runtime_error{
+                    "SymRefreshModuleList failed with " + IceInternal::errorToString(GetLastError())};
+            }
+        }
+    }
+    else
+    {
+        HANDLE currentProcess = GetCurrentProcess();
+        // duplicate handle as per https://learn.microsoft.com/en-us/windows/win32/debug/initializing-the-symbol-handler
+        if (!DuplicateHandle(currentProcess, currentProcess, currentProcess, &process, 0, FALSE, DUPLICATE_SAME_ACCESS))
+        {
+            throw std::runtime_error{
+                "DuplicateHandle on current process failed with " + IceInternal::errorToString(GetLastError())};
+        }
+
+        SymSetOptions(SYMOPT_LOAD_LINES | SYMOPT_DEFERRED_LOADS | SYMOPT_EXACT_SYMBOLS | SYMOPT_UNDNAME);
+        if (!SymInitialize(process, nullptr, TRUE))
+        {
+            CloseHandle(process);
```

Comment:
Maybe construct the error message before calling `CloseHandle` so we get the error from SymInitialize, and not from CloseHandle if it happens to fail too.

Reply:
You're right. It's likely CloseHandle resets the last error to 0. Fixed.